src/blueapi/service/model.py

```diff
@@ -112,3 +112,7 @@ class StateChangeRequest(BlueapiBaseModel):
         description="Should worker defer Pausing until the next checkpoint",
         default=False,
     )
+    reason: Optional[str] = Field(
```

Comment:
Current implementation is stopping/aborting the running task is by putting aborting/stopping to /worker/state and so it belongs here.

If we had Task State that supported Stopped/Aborted (~ #199 if we formalise the schema of task states, or put Stopped/Aborted/Reason in errors?), could be to /worker/task/state, or just updating /worker/task with a new state and it could be on WorkerTask. I invite thoughts on whether that's preferable:
I wanted cancelling a task to be DELETE /worker/task, because we make a task run by PUT /worker/task and it seems analogous: but DELETE is discouraged from having a body and the library we use for tests only allows it in a workaround, and we require Abort reason at least.

We already Pause/Resume a task by pausing/resuming the RunEngine, by requesting a change of the state of the Worker, which updates its state whenever the state of the RunEngine changes. And it seems again neat to Stop/Abort a task by calling stop/abort on the RunEngine, by requesting a change of the state of the Worker...

I think either pause/resume/stop/abort all belong on /worker/task/state, and we need to merge #199 and refactor #229 and this, or pause/resume/stop/abort belong on /worker/state and this is for now the correct place.

Reply:
I think they should all be on /worker/state as they are all to do with the state of the Run engine (i.e. worker) in which case the task element of the URL is redundant in evaluating that, as the there is only ever 0 or 1 running task. Also that would then lead to appropriate error conditions if no task was attached to the RE. In this model the /worker/task endpoint is for a different purpose - to see if there is one loaded into the worker and what it is using GET, or to attach one to the worker using PUT.